File: modules/fishing-lab-hands-free/src/FishingLabHandsFreeModule.ts
import { requireNativeModule } from 'expo';
import { WatchCompanionStatus } from '../../../src/types/handsFree';

type FishingLabHandsFreeModuleType = {
  consumePendingCommand(): Promise<string | null>;
  syncActiveOuting(serializedOuting: string | null): Promise<void>;
  syncHandsFreePreferences(serializedPreferences: string): Promise<void>;
  getWatchCompanionStatus(): Promise<WatchCompanionStatus>;
};

export default requireNativeModule<FishingLabHandsFreeModuleType>('FishingLabHandsFree');

File: modules/fishing-lab-hands-free/ios/FishingLabHandsFreeModule.swift
import ExpoModulesCore

public class FishingLabHandsFreeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("FishingLabHandsFree")

    let syncManager = FishingLabWatchSyncManager.shared

    AsyncFunction("consumePendingCommand") {
      syncManager.consumePendingCommand()
    }

    AsyncFunction("syncActiveOuting") { (serializedOuting: String?) in
      syncManager.syncActiveOuting(serializedOuting)
    }

    AsyncFunction("syncHandsFreePreferences") { (serializedPreferences: String) throws in
      try syncManager.syncHandsFreePreferences(serializedPreferences)
    }

    AsyncFunction("getWatchCompanionStatus") {
      syncManager.getWatchCompanionStatus()
    }
  }
}
